code/modules/events/supermatter_surge.dm

```diff
@@ -0,0 +1,110 @@
+	min_wizard_trigger_potency = 4
+	max_wizard_trigger_potency = 7
+	admin_setup = list(
+		/datum/event_admin_setup/input_number/surge_spiciness,
+	)
+
+/datum/round_event_control/supermatter_surge/can_spawn_event(players_amt, allow_magic = FALSE)
+	. = ..()
+
+	if(SSjob.is_skeleton_engineering(crew_threshold = 2))
+		return FALSE
+
+/datum/round_event/supermatter_surge
+	announce_when = 1
+	end_when = SUPERMATTER_SURGE_DURATION_MIN
+	/// How powerful is the supermatter surge going to be?
+	var/surge_class = SUPERMATTER_SURGE_SEVERITY_MIN
+	/// The default bullet energy of the crystal
+	var/starting_bullet_energy = SUPERMATTER_DEFAULT_BULLET_ENERGY
+	/// The default heat modifier of nitrogen
+	var/starting_heat_modifier = -2.5
+	/// The default powerloss inhibition of nitrogen
+	var/starting_powerloss_inhibition = 0
+	/// Typecasted reference to the supermatter chosen at event start
+	var/obj/machinery/power/supermatter_crystal/engine
+	/// Typecasted reference to the nitrogen properies in the SM chamber
+	var/datum/sm_gas/nitrogen/sm_gas
+
+/datum/event_admin_setup/input_number/surge_spiciness
+	input_text = "Set surge intensity. (Higher is more severe.)"
+	min_value = 1
+	max_value = 4
+
+/datum/event_admin_setup/input_number/surge_spiciness/prompt_admins()
+	default_value = rand(1, 4)
+	return ..()
+
+/datum/event_admin_setup/input_number/surge_spiciness/apply_to_event(datum/round_event/supermatter_surge/event)
+	event.surge_class = chosen_value
+
+/datum/round_event/supermatter_surge/setup()
+	engine = GLOB.main_supermatter_engine
+	if(isnull(engine))
+		stack_trace("SM surge event failed to find a supermatter engine!")
+		return
+
+	sm_gas = LAZYACCESS(GLOB.sm_gas_behavior, /datum/gas/nitrogen)
+	if(isnull(sm_gas))
+		stack_trace("SM surge event failed to find gas properties for the supermatter engine.")
+		return
+
+	starting_bullet_energy = engine.bullet_energy
+	starting_heat_modifier = sm_gas.heat_modifier
+	starting_powerloss_inhibition = sm_gas.powerloss_inhibition
+	if(isnull(surge_class))
+		surge_class = rand(SUPERMATTER_SURGE_SEVERITY_MIN, SUPERMATTER_SURGE_SEVERITY_MAX)
+
+	end_when = rand(SUPERMATTER_SURGE_DURATION_MIN + (surge_class * 45 EVENT_SECONDS), SUPERMATTER_SURGE_DURATION_MAX)
+
+/datum/round_event/supermatter_surge/announce()
+	priority_announce("CIMS has detected unusual atmospheric properties in the supermatter chamber, energy output from the supermatter crystal has shown a significant and unanticipated increase. Engineering intervention is required to stabilize the crystal.", "Supermatter Surge Alert: Class [surge_class]", 'sound/machines/engine_alert3.ogg')
+
+/datum/round_event/supermatter_surge/start()
+	engine.bullet_energy = SUPERMATTER_SURGE_BULLET_ENERGY_ADDITION + surge_class
+	sm_gas.heat_modifier += clamp(SUPERMATTER_SURGE_HEAT_MODIFIER * surge_class, 0.5, 2) // letting this get to 0 or higher will be a Bad Time
+	sm_gas.powerloss_inhibition = SUPERMATTER_SURGE_POWERLOSS_INHIBITION * surge_class
+
```

Comment:
I don't see anything that would stop the SM from just setting the heat_modifier or powerloss_inhibition back to their natural values.

The SM usually sets these sorts of values by summing up a bunch of values from a list of factors. Unfortunately the list's scope is in the proc, but maybe you can move those lists' scopes to the SM level. You could then add an SM surge factor for the heat modifier and powerloss inhibition at the start of the event, and remove these factors when the event ends.

Reply:
> I don't see anything that would stop the SM from just setting the heat_modifier or powerloss_inhibition back to their natural values.
> 
> It doesn't. It reads the datum for gas info.

Reply:
Oh I misread it.